core/src/main/scala/com/yahoo/maha/core/query/QueryPipeline.scala

```diff
@@ -847,18 +857,19 @@ OuterGroupBy operation has to be applied only in the following cases
 
     def runMultiEngineQuery(factBestCandidateOption: Option[FactBestCandidate], bestDimCandidates: SortedSet[DimensionBundle], queryGenVersion: Version): QueryPipelineBuilder = {
       val indexAlias = bestDimCandidates.last.publicDim.primaryKeyByAlias
+      val factGroupByKeys = requestModel.bestCandidates.get.dimColMapping.values.toList
```

Comment:
runMultiEngineQuery method is meant to run dim driven queries, not fact driven.

Reply:
The group by keys are there for the Fact subqueries.  I will modify the rendered Dimension queries to ignore this value.